### PR TITLE
Always apply all widget changes when saving aggregation builder.

### PR DESCRIPTION
**Pull request by linuspahl**
## Motivation and Context

Before this change we were only saving changes, when clicking on the “Save” button in the aggregation builder, which already have been applied by clicking the:
- search submit button, for widget search controls changes (on a dashboard)
- “Apply changes” button, for widget aggregation element changes.

With this PR we are always saving every valid search control and aggregation element change, when the user is saving the widget. Furthermore we:
- Renamed the aggregation elements “Apply Changes” button to “Update preview” and unified the background color with the search controls submit button.
- Renamed the widget edit “Save” button to “Apply Changes”

This way it is easier to understand what effect each button has.

Fixes: https://github.com/Graylog2/graylog2-server/issues/10560

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/views/components/contexts/WidgetEditApplyAllChangesProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      if (updatedWidgetConfig) {
+        newWidget = newWidget.toBuilder().config(updatedWidgetConfig).build();
+        hasChanges = hasChanges || true;
+      }
+    }
+
+    if (hasChanges) {
+      setIsSubmitting(true);
+
+      return WidgetActions.update(widget.id, newWidget)
```
Comment (linuspahl) [inline graylog2-web-interface/src/views/components/contexts/WidgetEditApplyAllChangesProvider.tsx:69]: When a user changes the widget and directly clicks on the "Apply changes" button we are currently executing the search twice. Do you have an idea what could be a good solution for this problem?

File: graylog2-web-interface/src/views/components/widgets/Widget.test.tsx [hunk 2]
```diff
@@ -45,6 +41,8 @@ jest.mock('../searchbar/QueryInput', () => mockComponent('QueryInput'));
 jest.mock('./WidgetHeader', () => 'widget-header');
 jest.mock('./WidgetColorContext', () => ({ children }) => children);
 
+// jest.mock('views/components/contexts/WidgetEditApplyAllChangesProvider', () => ({ children }) => <WidgetEditApplyAllChangesContext.Provider></>);
```
Comment (dennisoelkers) [inline graylog2-web-interface/src/views/components/widgets/Widget.test.tsx:44]: I guess this line can be removed?

**Review by dennisoelkers — CHANGES_REQUESTED** [review]
The PR needs a rebase before I can test it properly. I added two smaller annotations inline, maybe it makes sense to address them (or not) together with the rebase?

File: graylog2-web-interface/src/views/components/aggregationwizard/__tests__/AggregationWizard.groupBy.test.tsx [hunk 3]
```diff
@@ -94,7 +94,7 @@ describe('AggregationWizard', () => {
       await selectEvent.select(fieldSelection, 'took_ms');
     });
 
-    const applyButton = await screen.findByRole('button', { name: 'Apply Changes' });
+    const applyButton = await screen.findByRole('button', { name: 'Update Preview' });
```
[the remark this answers is not in the document]
Comment (linuspahl) [reply graylog2-web-interface/src/views/components/aggregationwizard/__tests__/AggregationWizard.groupBy.test.tsx:97]: Would you create one helper per file or one helper for all files?

[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog2-web-interface/src/views/components/aggregationwizard/__tests__/AggregationWizard.groupBy.test.tsx:97]: Good question. Both ways have their pros/cons. I guess one helper per file is a good compromise between keeping tests decoupled and allowing to make changes easily.

**Review by dennisoelkers — APPROVED** [review]
Looks great now! :tada:

**Pull request merged**